src/main/java/org/spongepowered/common/data/processor/data/item/HideDataProcessor.java

```diff
@@ -0,0 +1,160 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.item;
+
+import com.google.common.collect.Maps;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutableHideData;
+import org.spongepowered.api.data.manipulator.mutable.item.HideData;
+import org.spongepowered.common.data.manipulator.mutable.item.SpongeHideData;
+import org.spongepowered.common.data.processor.common.AbstractMultiDataSingleTargetProcessor;
+import org.spongepowered.common.data.util.NbtDataUtil;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class HideDataProcessor extends AbstractMultiDataSingleTargetProcessor<ItemStack, HideData, ImmutableHideData> {
+
+    public HideDataProcessor() {
+        super(ItemStack.class);
+    }
+
+    @Override
+    protected boolean doesDataExist(ItemStack dataHolder) {
+        return true;
+    }
+
+    @Override
+    protected boolean set(ItemStack dataHolder, Map<Key<?>, Object> keyValues) {
+        if (!dataHolder.hasTagCompound()) {
+            dataHolder.setTagCompound(new NBTTagCompound());
+        }
+        int flag = 0;
+        if ((boolean) keyValues.get(Keys.HIDE_ENCHANTMENTS)) {
+            flag |= 1;
+        }
+        if ((boolean) keyValues.get(Keys.HIDE_ATTRIBUTES)) {
+            flag |= 2;
+        }
+        if ((boolean) keyValues.get(Keys.HIDE_UNBREAKABLE)) {
+            flag |= 4;
+        }
+        if ((boolean) keyValues.get(Keys.HIDE_CAN_DESTROY)) {
+            flag |= 8;
+        }
+        if ((boolean) keyValues.get(Keys.HIDE_CAN_PLACE)) {
+            flag |= 16;
+        }
+        if ((boolean) keyValues.get(Keys.HIDE_MISCELLANEOUS)) {
+            flag |= 32;
+        }
+        dataHolder.getTagCompound().setInteger(NbtDataUtil.ITEM_HIDE_FLAGS, flag);
+        return true;
+    }
+
+    @Override
+    protected Map<Key<?>, ?> getValues(ItemStack dataHolder) {
+        if (!dataHolder.hasTagCompound()) {
+            return Maps.newHashMap();
+        }
+        Map<Key<?>, Boolean> map = Maps.newHashMap();
+        int flag = dataHolder.getTagCompound().getInteger(NbtDataUtil.ITEM_HIDE_FLAGS);
+        if ((flag & 32) != 0) {
+            map.put(Keys.HIDE_MISCELLANEOUS, true);
+            flag -= 32;
```

Comment:
I don't think it's necessary to subtract the value. Since all of the flags are powers of 2, you're only checking if one specific bit is set.